src/main/scala/io/qbeast/spark/internal/sources/v2/QbeastPartitionReaderFactory.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Qbeast Analytics, S.L.
+ */
+package io.qbeast.spark.internal.sources.v2
+
+import org.apache.hadoop.fs.FileStatus
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.connector.read.{
+  InputPartition,
+  PartitionReader,
+  PartitionReaderFactory
+}
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.execution.PartitionedFileUtil
+import org.apache.spark.sql.execution.datasources.{FileIndex, PartitionDirectory, PartitionedFile}
+import org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat
+import org.apache.spark.sql.types.StructType
+
+/**
+ * Implementation of [[PartitionReaderFactory]].
+ */
+class QbeastPartitionReaderFactory private[v2] (
```

Comment:
I think this class should also override the methods for creating a columnar reader (PartitionReader<ColumnarBatch> createColumnarReader(InputPartition partition)). 
In the PartitionReaderFactory interface, these methods return exceptions by default, but we should support columnar readying.

Reply:
Agree. There's also the method `supportColumnarReads` which tells the Plan that the Format supports columnar batch.